app/questionnaire/router.py

```diff
@@ -66,7 +66,8 @@ def get_first_incomplete_location_in_questionnaire_url(self) -> str:
         return self.get_next_location_url_for_end_of_section()
 
     def get_last_location_in_questionnaire_url(self) -> str:
-        routing_path = self.routing_path(*self._get_last_complete_section_key())
+        # Expects a vardic argument but can be None
+        routing_path = self.routing_path(*self._get_last_complete_section_key())  # type: ignore[misc]
         return self.get_last_location_in_section(routing_path).url()
```

Comment:
The comment should be about why we need to ignore it. The actual warning is genuine. This shouldn't error given it is currently only called from the submit handler (very end). But it would be better to fix this, if this was called by anything else, that wouldn't be true. Also this is used for `previous_location_url` which in the html handles not existing.

```py
    def get_last_location_in_questionnaire_url(self) -> Optional[str]:
        # Expects a vardic argument but can be None

        section_key = self._get_last_complete_section_key()
        if section_key:        
            routing_path = self.routing_path(*section_key)
            return self.get_last_location_in_section(routing_path).url()
```

Once this is changed, `get_previous_location_url` of submit questionnaires needs to change to `Optional[str]`.

Reply:
Changes added, also had to change `render_template` type hinting from `**kwargs: Union[str, Mapping]` to `**kwargs: Union[Optional[str], Mapping]`